### PR TITLE
Add docs for k8s floating IP support

### PR DESCRIPTION
**Pull request by caseydavenport**
## Description

Adds docs for https://github.com/projectcalico/cni-plugin/pull/605 and https://github.com/projectcalico/cni-plugin/pull/605

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

**Comment by tomdee** [comment]
Deploy preview for *calico* ready!

Built with commit d237be1542002bfc90bce2a4c9da5259f821c1ac

https://deploy-preview-2213--calico.netlify.com

**Comment by tomdee** [comment]
Deploy preview for *calico* ready!

Built with commit 4b9562a194eed4fc2facbcc329d261452e0c8b61

https://deploy-preview-2213--calico.netlify.com

File: master/reference/cni-plugin/configuration.md [hunk 1]
```diff
@@ -335,3 +335,36 @@ There are two annotations to request a specific IP address:
...
+
+You can request a floating IP address for a pod through [Kubernetes annotations](https://kubernetes.io/docs/user-guide/annotations/) with {{site.prodname}}.
+
+- `cni.projectcalico.org/floatingIPs`: A list of floating IPs which will be assigned to the pod's workload endpoint.
+
+  Example:
+
+   ```yaml
+   annotations:
+        "cni.projectcalico.org/floatingIPs": "[\"10.0.0.1\"]"
```
Comment (neiljerram) [inline master/reference/cni-plugin/configuration.md]: The quoting and escaping here looks surprisingly complex for YAML.  Are you sure that it is correct?

Comment (caseydavenport) [reply master/reference/cni-plugin/configuration.md]: Yeah, the value here is a `json` structure represented as a yaml string.

We could probably do something like this instead, though using mixed quotes: `'["10.0.0.1"]'`

Comment (neiljerram) [reply master/reference/cni-plugin/configuration.md]: I'm happy as is.

File: master/reference/cni-plugin/configuration.md [hunk 2]
```diff
@@ -335,3 +335,36 @@ There are two annotations to request a specific IP address:
 > - You can only specify one IPv4/IPv6 or one IPv4 and one IPv6 address with these annotations.
 > - When `ipAddrs` or `ipAddrsNoIpam` is used with `ipv4pools` or `ipv6pools`, `ipAddrs` / `ipAddrsNoIpam` take priority.
 {: .alert .alert-info}
+
+#### Requesting a floating IP
+
+You can request a floating IP address for a pod through [Kubernetes annotations](https://kubernetes.io/docs/user-guide/annotations/) with {{site.prodname}}.
```
Comment (bradbeam) [inline master/reference/cni-plugin/configuration.md]: A little late to the convo on this, but should we mention that the floating IP must be part of a defined calico pool for it to be advertised properly?

Comment (caseydavenport) [reply master/reference/cni-plugin/configuration.md]: Yes, that is a good point. We should update the doc to mention that.

**Pull request merged**